paparazzi/paparazzi/src/main/java/app/cash/paparazzi/accessibility/A11ySnapshotHandler.kt

```diff
@@ -0,0 +1,231 @@
+          paint = colorForIndex(i)
+          stroke = BasicStroke(3f)
+          val element = it.scaleBy(scale)
+          drawRect(
+            element.displayBounds.left,
+            element.displayBounds.top,
+            element.displayBounds.width(),
+            element.displayBounds.height()
+          )
+          paint = Color(color.red, color.green, color.blue, 255 / 4)
+          fillRect(
+            element.displayBounds.left,
+            element.displayBounds.top,
+            element.displayBounds.width(),
+            element.displayBounds.height()
+          )
+          if (element.touchBounds != null) {
+            drawRect(
+              element.touchBounds.left,
+              element.touchBounds.top,
+              element.touchBounds.width(),
+              element.touchBounds.height()
+            )
+          }
+        }
+      }
+    }
+
+    internal fun drawLegend(
+      accessibilityState: AccessibilityState,
+      image: BufferedImage
+    ): BufferedImage {
+      val modifiedImage = BufferedImage(600, image.height, image.type)
+
+      return modifiedImage.withGraphics2D {
+        paint = Color.WHITE
+        fillRect(0, 0, modifiedImage.width, modifiedImage.height)
+
+        font = font.deriveFont(20f)
+        stroke = BasicStroke(3f)
+
+        var index = 1
+
+        fun drawItem(s: String) {
+          drawString(s, 50f, 28f * index++)
+        }
+
+        accessibilityState.elements.forEachIndexed { i, it ->
+          paint = Color.BLACK
+
+          val start = index
+          if (it.role != null || it.disabled || it.heading) {
+            val role = if (it.role != null) "Role " + it.role + " " else ""
+            val heading = if (it.heading) "Heading " else ""
+            val disabled = if (it.disabled) "Disabled" else ""
+            drawItem(role + heading + disabled)
+          }
+          if (it.contentDescription != null) {
+            drawItem("Content Description \"${it.contentDescription.joinToString(", ")}\"")
+          } else if (it.text != null) {
+            drawItem("Text \"${it.text.joinToString(", ")}\"")
+          }
+          if (it.stateDescription != null) {
+            drawItem("State Description \"${it.stateDescription}\"")
+          }
+          if (it.onClickLabel != null) {
+            drawItem("On Click \"${it.onClickLabel}\"")
+          }
+          if (it.progress != null) {
+            drawItem("Progress \"${it.progress}\"")
+          }
+          if (it.customActions != null) {
+            it.customActions.forEach {
+              drawItem("Custom Action \"${it.label}\"")
+            }
+          }
+          val end = index
+
+          paint = colorForIndex(i)
+          drawRect(10, start * 28 - 21, modifiedImage.width - 20, (end - start) * 28)
+
+          index++
+        }
+      }
+    }
+
+    private fun colorForIndex(i: Int): Color {
+      return colors[i % colors.size]
+    }
```

Comment:
nit: We had an issue deciding colors this way as it's not consistent across updating UI thereby bumping the index for everything. Instead, we opted for using the view id to ensure the most consistent color for views.

Reply:
I will take a look at what to do here. We don't have ids, but I'm sure I can find something similar.